swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -300,14 +300,14 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
                                  forResult: apiClient.track(inAppClick: messageId, clickedUrl: clickedUrl, deviceMetadata: deviceMetadata))
     }
     
-    func trackInAppClick(_ message: IterableInAppMessage, location: InAppLocation = .unknown, clickedUrl: String) {
+    func trackInAppClick(_ message: IterableInAppMessage, location: InAppLocation = .inApp, clickedUrl: String) {
         IterableAPIInternal.call(successHandler: IterableAPIInternal.defaultOnSucess(identifier: "trackInAppClick"),
                                  andFailureHandler: IterableAPIInternal.defaultOnFailure(identifier: "trackInAppClick"),
-                                 forResult: apiClient.track(inAppClick: InAppMessageContext(message: message, location: location, deviceMetadata: deviceMetadata), clickedUrl: clickedUrl))
+                                 forResult: apiClient.track(inAppClick: InAppMessageContext.from(message: message, location: location, deviceMetadata: deviceMetadata), clickedUrl: clickedUrl))
     }
     
-    func trackInAppClose(_ message: IterableInAppMessage, location: InAppLocation = .unknown, source: InAppCloseSource = .unknown, clickedUrl: String? = nil) {
-        let result = apiClient.track(inAppClose: InAppMessageContext(message: message, location: location, deviceMetadata: deviceMetadata),
+    func trackInAppClose(_ message: IterableInAppMessage, location: InAppLocation = .inApp, source: InAppCloseSource = .other, clickedUrl: String? = nil) {
```

Comment:
Can we make these parameters required and remove `InAppCloseSource.other`? How would a developer use `InAppCloseSource.other`?